cmd/root.go

```diff
@@ -60,38 +68,89 @@ func initConfig() {
 	}
 }
 
+var (
+	eventsCounter = prometheus.NewCounter(prometheus.CounterOpts{
+		Namespace: "ssl",
+		Subsystem: "agent",
+		Name:      "events",
+		Help:      "Number of events handled.",
+	})
+
+	eventsFatalErrorCounter = prometheus.NewCounter(prometheus.CounterOpts{
+		Namespace: "ssl",
+		Subsystem: "agent",
+		Name:      "fatal_error_requests",
+		Help:      "Number of events causing fatal errors",
+	})
+
+	eventsErrorCounter = prometheus.NewCounter(prometheus.CounterOpts{
+		Namespace: "ssl",
+		Subsystem: "agent",
+		Name:      "error_events",
+		Help:      "Number of events in error",
+	})
+
+	eventsKafkaConsumerErrorCounter = prometheus.NewCounter(prometheus.CounterOpts{
+		Namespace: "ssl",
+		Subsystem: "agent",
+		Name:      "kafka_error_events",
+		Help:      "Number of events causing kafka consumer error",
+	})
+
+	eventsSchedulerErrorCounter = prometheus.NewCounter(prometheus.CounterOpts{
+		Namespace: "ssl",
+		Subsystem: "agent",
+		Name:      "scheduler_error_events",
+		Help:      "Number of events causing scheduler error",
+	})
+
+	sslHaltedCounter = prometheus.NewCounter(prometheus.CounterOpts{
+		Namespace: "ssl",
+		Subsystem: "agent",
+		Name:      "halts",
+		Help:      "Number of ssl agent  halts",
+	})
+)
+
 // RootCmd launch the aggregator agent.
 var RootCmd = &cobra.Command{
 	Use:   "poke-ssl-agent",
 	Short: "poke-ssl-agent collect SSL domains stats",
 	Run: func(cmd *cobra.Command, args []string) {
 		log.Info("poke-ssl-agent starting")
 
+		http.Handle("/metrics", promhttp.Handler())
+		http.ListenAndServe(":2112", nil)
+
 		quit := make(chan os.Signal, 2)
 		signal.Notify(quit, syscall.SIGTERM)
 		signal.Notify(quit, syscall.SIGINT)
 
 		consumer, err := core.NewConsumer()
 		if err != nil {
+			eventsFatalErrorCounter.Inc()
 			log.WithError(err).Fatal("Could not start the kafka consumer")
 		}
 
 		for {
 			select {
 			case m, ok := <-consumer.Messages():
 				if !ok {
+					eventsFatalErrorCounter.Inc()
```

Comment:
if you fatal here, the metric you increment just before, won't have the time to be scrapped, you will never see this one